pydicom/encoders/base.py

```diff
@@ -283,7 +283,10 @@ def _encode_dataset(
         """Return a single encoded frame from the *Pixel Data* in `ds`."""
         kwargs.update(self.kwargs_from_ds(ds))
 
-        tsyntax = cast(UID, ds.file_meta.TransferSyntaxUID)
+        file_meta = (
+            cast("FileMetaDataset", ds.file_meta)  # type: ignore[has-type]
```

Comment:
Would prefer we define a property `Dataset.file_meta: Optional[FileMetaDataset] = None`.

But that's about half the reason for the existence of `FileDataset` right there.